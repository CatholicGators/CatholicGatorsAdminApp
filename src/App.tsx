import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { listenForUser, googleSignIn, signOut } from './redux/actions/auth/authActions';
import { connect } from 'react-redux';

type Props = {
  user: any;
  listenForUser: () => any;
}

class App extends Component<Props> {
  componentDidMount() {
    this.props.listenForUser();
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth.user
});

const mapDispatchToProps = dispatch => ({
  listenForUser: () => dispatch(listenForUser()),
  googleSignIn: () => dispatch(googleSignIn()),
  signOut: () => dispatch(signOut())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
