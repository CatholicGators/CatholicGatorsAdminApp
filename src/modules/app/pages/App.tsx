import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";

import { listenForUser } from '../../../redux/actions/auth/authActions';
import Header from '../components/Header/Header';
import Admin from '../../admin/pages/Admin';
import AuthorizedRoute from '../components/AuthorizedRoute/AuthorizedRoute';
import ContactForm from '../../contactForm/pages/ContactForm/ContactForm'
import MyContacts from '../../contactForm/pages/MyContacts/MyContacts';

type Props = {
  user: any;
  listenForUser: () => any;
}

export class App extends Component<Props> {
  componentDidMount() {
    this.props.listenForUser();
  }

  render() {
    return (
      <Router>
        <div>
          <Header></Header>
          <Switch>
            <Route exact path='/' component={ContactForm}/>
            <AuthorizedRoute
              path="/admin/"
              component={Admin}
              isAuthorized={this.props.user ? this.props.user.isAdmin && this.props.user.isApproved : false}
              redirectPathname="/"
            />
            <Route path="/my-contacts/" component={MyContacts}/>
            <Route render={() => <Redirect to="/" />} />
          </Switch>
        </div>
      </Router>
    );
  }
}

const mapStateToProps = state => ({
  user: state.auth.user
})

const mapDispatchToProps = dispatch => ({
  listenForUser: () => dispatch(listenForUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
