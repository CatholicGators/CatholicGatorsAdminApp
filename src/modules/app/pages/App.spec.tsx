import React from 'react'
import { shallow } from 'enzyme';

import { App } from './App';

describe('App', () => {
    let props, wrapper;

    beforeEach(() => {
        props = {
            listenForUser: jest.fn()
        };
        wrapper = shallow(<App {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should call listenForUser when mounted', () => {
        expect(props.listenForUser).toHaveBeenCalled();
    });
});