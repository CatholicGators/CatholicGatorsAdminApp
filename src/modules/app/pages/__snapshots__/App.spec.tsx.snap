// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    listenForUser={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <withRouter(Connect(WithStyles(Header))) />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <PrivateRoute
            component={[Function]}
            isAuthorized={false}
            path="/admin/"
            redirectPathname="/"
          />
          <Route
            render={[Function]}
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(WithStyles(Header))) />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <PrivateRoute
              component={[Function]}
              isAuthorized={false}
              path="/admin/"
              redirectPathname="/"
            />
            <Route
              render={[Function]}
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <PrivateRoute
                component={[Function]}
                isAuthorized={false}
                path="/admin/"
                redirectPathname="/"
              />,
              <Route
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "isAuthorized": false,
                "path": "/admin/",
                "redirectPathname": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/my-contacts/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <withRouter(Connect(WithStyles(Header))) />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <PrivateRoute
              component={[Function]}
              isAuthorized={false}
              path="/admin/"
              redirectPathname="/"
            />
            <Route
              render={[Function]}
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(WithStyles(Header))) />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <PrivateRoute
                component={[Function]}
                isAuthorized={false}
                path="/admin/"
                redirectPathname="/"
              />
              <Route
                render={[Function]}
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <PrivateRoute
                  component={[Function]}
                  isAuthorized={false}
                  path="/admin/"
                  redirectPathname="/"
                />,
                <Route
                  render={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "isAuthorized": false,
                  "path": "/admin/",
                  "redirectPathname": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/my-contacts/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
